contrib/src/main/java/com/datatorrent/contrib/parser/CsvParser.java

```diff
@@ -62,247 +71,193 @@
+  @Override
+  public KeyValPair<String, String> processErorrTuple(byte[] input)
   {
-    return fieldDelimiter;
+    //This method is not invoked for CSV parser
+    return null;
   }
 
   /**
-   * Sets the delimiter which separates fields in incoming data.
-   * 
-   * @param fieldDelimiter
+   * Returns array of cellprocessors, one for each field
    */
-  public void setFieldDelimiter(int fieldDelimiter)
+  private CellProcessor[] getProcessor(List<Field> fields)
   {
-    this.fieldDelimiter = fieldDelimiter;
+    CellProcessor[] processor = new CellProcessor[fields.size()];
+    int fieldCount = 0;
+    for (Field field : fields) {
+      processor[fieldCount++] = CellProcessorBuilder.getCellProcessor(field.getType(), field.getConstraints());
+    }
+    return processor;
   }
 
-  /**
-   * Gets the delimiter which separates lines in incoming data.
-   * 
-   * @return lineDelimiter
-   */
-  public String getLineDelimiter()
+  @Override
+  public void teardown()
   {
-    return lineDelimiter;
+    try {
+      csvMapReader.close();
+    } catch (IOException e) {
+      logger.error("Error while closing csv map reader {}", e.getMessage());
+    }
+    try {
+      csvBeanReader.close();
+    } catch (IOException e) {
+      logger.error("Error while closing csv bean reader {}", e.getMessage());
+    }
   }
 
   /**
-   * Sets the delimiter which separates line in incoming data.
+   * Complete hdfs path of schema
    * 
-   * @param lineDelimiter
+   * @return
    */
-  public void setLineDelimiter(String lineDelimiter)
+  public String getSchemaPath()
   {
-    this.lineDelimiter = lineDelimiter;
+    return schemaPath;
   }
 
   /**
-   * Gets the name of the fields with type and format ( for date ) as comma
-   * separated string in same order as incoming data. e.g
-   * name:string,dept:string,eid:integer,dateOfJoining:date|dd/mm/yyyy
+   * Set path of schema
    * 
-   * @return fieldInfo
+   * @param schemaPath
+   *          path of the schema file in hdfs
    */
-  public String getFieldInfo()
+  public void setSchemaPath(String schemaPath)
   {
-    return fieldInfo;
+    this.schemaPath = schemaPath;
+    try {
+      this.schema = FileUtils.readFromHDFS(schemaPath);
+    } catch (IOException e) {
+      DTThrowable.rethrow(e);
+    }
   }
 
   /**
-   * Sets the name of the fields with type and format ( for date ) as comma
-   * separated string in same order as incoming data. e.g
-   * name:string,dept:string,eid:integer,dateOfJoining:date|dd/mm/yyyy
-   * 
-   * @param fieldInfo
+   * output port to emit validate records as map
    */
-  public void setFieldInfo(String fieldInfo)
-  {
-    this.fieldInfo = fieldInfo;
-  }
-
-  private static final Logger logger = LoggerFactory.getLogger(CsvParser.class);
+  public final transient DefaultOutputPort<Map<String, Object>> parsedOutput = new DefaultOutputPort<Map<String, Object>>();
+  private static final Logger logger = LoggerFactory.getLogger(Parser.class);
```

Comment:
Not sure if there is any convention here.. But I have seen most of the files declaring the variables are the top of the file.